src/SumoCoders/FrameworkCoreBundle/Mail/MessageFactory.php

```diff
@@ -0,0 +1,253 @@
+     * @param string|null $html
+     * @param string|null $alternative
+     * @return \Swift_Message
+     */
+    protected function createMessage($subject = null, $html = null, $alternative = null)
+    {
+        $message = $this->createDefaultMessage();
+
+        if ($subject != '') {
+            $message->setSubject($subject);
+        }
+
+        if ($html != '' && $alternative != '') {
+            $message->setBody(
+                $this->wrapInTemplate($html),
+                'text/html'
+            );
+            $message->addPart($alternative, 'text/plain');
+        } elseif ($html != '' && $alternative == '') {
+            $message->setBody(
+                $this->wrapInTemplate($html),
+                'text/html'
+            );
+            $message->addPart(
+                $this->convertToPlainText($html),
+                'text/plain'
+            );
+        } elseif ($html == '' && $alternative != '') {
+            $message->setBody($alternative, 'text/plain');
+        }
+
+        return $message;
+    }
+
+    /**
+     * Create a HTML message
+     *
+     * If no alternative is provided it will be generated automatically.
+     * This is just an alias for createMessage
+     *
+     * @param string|null $subject
+     * @param string|null $html
+     * @param string|null $plainText
+     * @return \Swift_Message
+     */
+    public function createHtmlMessage($subject = null, $html = null, $plainText = null)
+    {
+        return $this->createMessage($subject, $html, $plainText);
+    }
+
+    /**
+     * Create a plain text message
+     *
+     * @param string|null $subject
+     * @param string|null $body
+     * @return \Swift_Message
+     */
+    public function createPlainTextMessage($subject = null, $body = null)
+    {
+        return $this->createMessage($subject, null, $body);
+    }
+
+    /**
+     * @return \Swift_Message
+     */
+    public function createDefaultMessage()
+    {
+        $message = \Swift_Message::newInstance();
+
+        if (!empty($this->sender)) {
+            $message->setFrom($this->sender);
+        }
+
+        if (!empty($this->replyTo)) {
+            $message->setReplyTo($this->replyTo);
+        }
+
+        if (!empty($this->to)) {
+            $message->setTo($this->to);
+        }
+
+        return $message;
+    }
+
+    /**
+     * Wrap the given content in a nice default email template
+     *
+     * @param string $content
+     * @return string
+     */
+    public function wrapInTemplate($content)
+    {
+        $css = file_get_contents($this->cssPath);
+        $html = $this->template->render(
+            $this->templatePath,
+            array(
+                'content' => $content,
+                'css' => $css,
+            )
+        );
+
+        $cssToInlineStyles = new CssToInlineStyles(
+            $html,
+            $css
+        );
+
+        $content = $cssToInlineStyles->convert();
+
+        return $content;
```

Comment:
no need to create this content var, you can directly return from the convert call;